crates/rust-analyzer/src/world.rs

```diff
@@ -38,6 +38,7 @@ pub struct Options {
     pub inlay_hints: InlayHintsOptions,
     pub rustfmt_args: Vec<String>,
     pub cargo_watch: CheckOptions,
+    pub diagnostics_config: DiagnosticsOptions,
```

Comment:
```suggestion
    pub diagnostics: DiagnosticsOptions,
```
I suggest to follow the pattern we have (i.e. not adding `config` suffix to the properties of this struct)